akka-distributed-data/src/main/scala/akka/cluster/ddata/ORMap.scala

```diff
@@ -174,8 +303,123 @@ final class ORMap[A, B <: ReplicatedData] private[akka] (
         case (None, None) ⇒ throw new IllegalStateException(s"missing value for $key")
       }
     }
+    val removedKeysIterator = (tombstoneVals.keys ++ that.tombstoneVals.keys).iterator
+    var removedValues = Map.empty[A, B]
+    removedKeysIterator.foreach { key ⇒
+      (this.tombstoneVals.get(key), that.tombstoneVals.get(key)) match {
+        case (Some(thisValue), Some(thatValue)) ⇒
+          if (thisValue.getClass != thatValue.getClass) {
+            val errMsg = s"Wrong type for merging [$key] in [${getClass.getName}], existing type " +
+              s"[${thisValue.getClass.getName}], got [${thatValue.getClass.getName}]"
+            throw new IllegalArgumentException(errMsg)
+          }
+          val removedValue = thisValue.merge(thatValue.asInstanceOf[thisValue.T]).asInstanceOf[B]
+          removedValues = removedValues.updated(key, removedValue)
+        case (Some(thisValue), None) ⇒
+          removedValues = removedValues.updated(key, thisValue)
+        case (None, Some(thatValue)) ⇒
+          removedValues = removedValues.updated(key, thatValue)
+        case (None, None) ⇒ throw new IllegalStateException(s"missing value for $key")
+      }
+    }
+    // tombstone the values for removed keys
```

Comment:
I will remove the tombstones and revert the vanilla `ORMap` to `delta key` - `full value` setup. However, as `ORMultiMap` already does the _clear_ trick:
```
  @InternalApi private[akka] def put(node: UniqueAddress, key: A, value: Set[B]): ORMultiMap[A, B] = {
    val newUnderlying = underlying.updated(node, key, ORSet.empty[B]) { existing ⇒
      value.foldLeft(existing.clear(node)) { (s, element) ⇒ s.add(node, element) }
    }
    new ORMultiMap(newUnderlying)
  }
```
I will keep it on `delta` - `delta` scheme, which for that case will not require tombstones and will not impact semantics.